### PR TITLE
Backend refactoring

### PR DESCRIPTION
**Pull request by ho-stanley**
All important endpoint works now.
Readme API documentation is not yet up to date but I thought it would still be good to deploy the working code as soon as possible and have it available for the frontend team.

### CODE REVIEW

File: backend/README.md [hunk 1]
```diff
@@ -0,0 +1,202 @@
...
+
+## API Endpoints
+
+### Public routes
+
+---
+
+### Signup
+
+Method: `POST`<br>
```
Comment (axelra82) [inline backend/README.md:33]: <br>? 🤔

File: backend/README.md [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+    -   `street` Street address - max length 250
+    -   `city` City name - max length 250
+    -   `zip` Zip code - max length 20
+
+#### Example of API response
+
+```js
+{
+	"status": 200,
+	"token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJlbWFpbCI6Im1AbS5jb20iLCJpYXQiOjE2MjE5MzQ1NTI4NjcsImV4cCI6MTYyMTkzODE1Mjg2N30.QnAfjfydKa4-Z2aLRuNFw8co6G9Oev8TEu3Aiyt3QVE",
```
Comment (axelra82) [inline backend/README.md]: Think about not showing sequences or other possible data that could in some form help potential hackers.

File: backend/models/address.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+		type: String,
+		maxlength: 250,
+	},
+	city: {
+		type: String,
+		maxlength: 250,
+	},
+	zip: {
+		type: Number,
+		max: 20,
```
Comment (axelra82) [inline backend/models/address.js]: Look in to `max` vs `maxlength` and `String` vs `Number` for `zip`

File: backend/auth/auth.js [hunk 4]
```diff
@@ -130,40 +71,135 @@ exports.userById = async (req, res, next) => {
...
  */
-exports.ratings = async (req, res, next) => {
+exports.updateUser = async (req, res, next) => {
+	const token = req.headers.authorization.split(' ')[1];
+	const payload = jwt.tokenPayload(token);
+	const time = new Date().getTime();
+
+	if (time < payload.exp) {
+		const updateUserId = req.params.id;
+		const id = req.user._id.toString();
```
Comment (axelra82) [inline backend/auth/auth.js:83]: `_id ` is unknown to frontend. User model does not contain `_id`, hence user can't set rating. Currently email (available in JWT) is unique identifier used in frontend.

Frontend and backend need to be unambiguous on how to identify users.

**Review by axelra82 — CHANGES_REQUESTED** [review]
~~strike comment~~

File: backend/README.md (hunk 1 above)
Comment (ho-stanley) [reply backend/README.md:33]: It's to add a new line directly below. I could have used two spaces but thought it was not clear when viewing the file.

File: backend/README.md (hunk 2 above)
Comment (ho-stanley) [reply backend/README.md]: Noted and changed.

File: backend/models/address.js (hunk 3 above)
Comment (ho-stanley) [reply backend/models/address.js]: Noted and fixed.

**Pull request merged**